lua/telescope/builtin/git.lua

```diff
@@ -349,11 +351,21 @@ git.status = function(opts)
 end
 
 local set_opts_cwd = function(opts)
+  local use_per_file_cwd = utils.get_default(opts.use_per_file_cwd, false)
   if opts.cwd then
     opts.cwd = vim.fn.expand(opts.cwd)
+  elseif use_per_file_cwd then
+    -- Run git commands in the context of the current file's repository.
+    -- A per-buffer git repository is used when finding files when used in conjunction
+    -- with the default use_git_root option.
+    opts.cwd = vim.fn.expand('%:h:p')
```

Comment:
I could not find a lua native way to do get the directory of the current buffer which is why it's doing `%:h:p`.

`vim.api.nvim_buf_get_dir(0)` would have been useful if it existed. I can get the filename but we would need to `os.path.dirname(_)` it. This works for me, but if there's a better way to get that I'm all ears.